lib/action_view/component/render_monkey_patch.rb

```diff
@@ -15,7 +15,7 @@ def render(options = {}, args = {}, &block)
 
           options.render_in(self, &block)
         elsif options.is_a?(Class) && options < ActionView::Component::Base
-          options.new(args).render_in(self, &block)
+          options.new(**args).render_in(self, &block)
         elsif options.is_a?(Hash) && options.has_key?(:component)
           options[:component].new(options[:locals]).render_in(self, &block)
         elsif options.respond_to?(:to_component_class) && !options.to_component_class.nil?
```

Comment:
If we want to fix this we need to solve it for the other two cases where we call `new`.

How about we explicitly check the arity of the constructor?

```
def new_component(component_class, options)
  arity = component_class.method(:initialize).arity
  if arity == 0 ? component_class.new : component_class.new(options)
end
```

and then use it in the if-else block `new_component(options, args).render_in(self, &block)` etc. etc.

Reply:
needs test cases for the other branches of the if-else

Reply:
Oh yeah, good point! Thanks for taking a look at this. I’ll add the other branches and see how it looks. I have a feeling that the airity check that you suggested will be simpler for the other cases.

 I guess the only thing we’d want to avoid is not dropping arguments (incorrectly) provided by the user to a 0-arity function. As it might make for a slightly confusing debugging experience. But \_(ツ)_/¯ it’s probably ok if we do. 

Also, maybe it’s as simple as something like `if arity ==0 && options.blank? ...` to avoid most cases.